src/mgr/PyModules.cc

```diff
@@ -544,8 +544,13 @@ void PyModules::set_config(const std::string &handle,
   }
   set_cmd.wait();
 
-  // FIXME: is config-key put ever allowed to fail?
-  assert(set_cmd.r == 0);
+  if (set_cmd.r != 0) {
+    // config-key put will fail if mgr's auth key has insufficient
+    // permission to set config keys
+    // FIXME: should this somehow raise an exception back into Python land?
+    dout(0) << "`config-key put " << global_key << " " << val << "` failed: "
+      << cpp_strerror(set_cmd.r) << dendl;
```

Comment:
@tserong i don't think we can assume that monitor does not provide any more helpful information than "access denied". so i think it would helpful if we can have all available error message in mgr's log file even it sounds redundant: error 13 versus `error 13 + "access denied". BTW, i suggested we print the errorno along with the error message. not just the error message from mon.